homeassistant/components/plex/__init__.py

```diff
@@ -1 +1,195 @@
+                configurator = hass.components.configurator
+                configurator.request_done(request_id)
+                _LOGGER.debug("Discovery configuration done")
+            if configurator_info:
+                # Write plex.conf if created via discovery/configurator
+                save_json(
+                    hass.config.path(PLEX_CONFIG_FILE),
+                    {
+                        host_and_port: {
+                            CONF_TOKEN: server_config[CONF_TOKEN],
+                            CONF_SSL: use_ssl,
+                            "verify": server_config[CONF_VERIFY_SSL],
+                        }
+                    },
+                )
+
+            if not hass.data.get(PLEX_MEDIA_PLAYER_OPTIONS):
+                hass.data[PLEX_MEDIA_PLAYER_OPTIONS] = {
+                    CONF_USE_EPISODE_ART: False,
+                    CONF_SHOW_ALL_CONTROLS: False,
+                    CONF_REMOVE_UNAVAILABLE_CLIENTS: True,
+                    CONF_CLIENT_REMOVE_INTERVAL: 600,
+                }
+
+            for platform in PLATFORMS:
+                hass.helpers.discovery.load_platform(
+                    platform, PLEX_DOMAIN, {}, original_config
+                )
+
+    def request_configuration(host_and_port):
+        """Request configuration steps from the user."""
+        configurator = hass.components.configurator
+        if host_and_port in _CONFIGURING:
+            configurator.notify_errors(
+                _CONFIGURING[host_and_port], "Failed to register, please try again."
+            )
+            return
+
+        def plex_configuration_callback(data):
+            """Handle configuration changes."""
+            config = {
+                "host_and_port": host_and_port,
+                CONF_TOKEN: data.get("token"),
+                CONF_SSL: cv.boolean(data.get("ssl")),
+                CONF_VERIFY_SSL: cv.boolean(data.get("verify_ssl")),
+            }
+            setup_plex(configurator_info=config)
+
+        _CONFIGURING[host_and_port] = configurator.request_config(
+            "Plex Media Server",
+            plex_configuration_callback,
+            description="Enter the X-Plex-Token",
+            entity_picture="/static/images/logo_plex_mediaserver.png",
+            submit_caption="Confirm",
+            fields=[
+                {"id": "token", "name": "X-Plex-Token", "type": ""},
+                {"id": "ssl", "name": "Use SSL", "type": ""},
+                {"id": "verify_ssl", "name": "Verify SSL", "type": ""},
+            ],
+        )
+
+    # End of inner functions.
+
+    if config:
+        original_config = config
+
+    hass.data.setdefault(PLEX_DOMAIN, {SERVERS: {}})
+
+    if hass.data[PLEX_DOMAIN][SERVERS]:
+        _LOGGER.debug("Plex server already configured")
+        return False
+
+    if PLEX_DOMAIN not in config:
+        discovery.listen(hass, SERVICE_PLEX, server_discovered)
+        return True
+
+    setup_plex(config=config)
```

Comment:
We should check the return value here and return False in turn if the return value is False.

Reply:
Updated. I think that should work.